### ISSUE TITLE
Status bar Not Accessible with Screen Reader on MacOS and Linux.

### ISSUE THREAD

**Issue by chigkim** [wxWidgets/wxWidgets#24305]
Screen readers can access status bar on Windows.
However, VoiceOver (screenreader on MacOS) and Orca (screenreader on Linux) cannot access status bar.
I found the same issue on WXPython.
https://github.com/wxWidgets/Phoenix/issues/1507

**Comment by vadz** [comment]
The problem is that for non-MSW platforms `wxStatusBar` uses a "generic" implementation, i.e. it's just a window drawn by wxWidgets itself and so screenreaders don't know anything about it.

So the possible solutions are

1. Replace it with a native control, if it exists, but I don't think this is the case under Mac/Cocoa (`NSStatusBar` is something quite different).
2. Provide generic a11y support for generic windows in wx.

Both would be nice and (2) would be especially so, but it's a lot of work and we don't have nearly enough resources to embark on a project like this, unfortunately, so it's not going to happen without outside contribution, sorry.

**Comment by chigkim** [comment]
Thanks for the reply. I'm not sure how QT does it, but I've seen QT Apps 
with status bar accessible on both MacOS as well as Windows.

**Comment by oneeyeman1** [comment]
@chigkim , QT does not use native controls and it is backed up by big coropration.
Thank you.

**Comment by robin24** [comment]
Hi,

I'm a blind developer myself and just came across this bug after hitting the inaccessible status bar issue myself.
I'm very new to wxPython and started looking into it for cross-platform UI development, the sizers are super handy from a blind UI dev perspective as they make it significantly easier to lay out views in a meaningful way.
However, it does look like accessibility support for VoiceOver on macOS does leave a lot to be desired. For instance, I've just discovered that wx.ListCtrl also isn't accessible, VoiceOver is completely unable to focus on this control.

I understand that it is necessary for wx to draw certain UI controls in some cases, and that it can be challenging sometimes to replace those with native UI components.
However, macOS has a very powerful accessibility API allowing for programmatic modification of accessibility labels, values and traits, enabling developers to emulate native UI behavior for VoiceOver.
I'm not familiar enough with wxPython's architecture to tell whether this might be a feasible approach, but just thought I'd put it out there as a suggestion.

**Comment by vadz** [comment]
As mentioned in [a previous comment](https://github.com/wxWidgets/wxWidgets/issues/24305#issuecomment-1937973458) it would be definitely great to implement a11y support for the generic controls and any contributions doing this would be very, very welcome. If you're familiar with Apple a11y API, I think implementing it for a simple control like `wxStatusBar` should be pretty simple. `wxListCtrl` would be more challenging -- in this case replacing it with (native) `wxDataViewCtrl` might be a better approach.
